packages/tools/src/utilities/contours/getDeduplicatedVTKPolyDataPoints.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Iterate through polyData from vtkjs and merge any points that are the same
+ * then update merged point references within lines array
+ * @param polyData - vtkPolyData
+ * @param bypass - bypass the duplicate point removal
+ * @returns the updated polyData
+ */
+export function getDeduplicatedVTKPolyDataPoints(polyData, bypass = false) {
+  const points = polyData.getPoints();
+  const lines = polyData.getLines();
+
+  const pointsArray = new Array(points.getNumberOfPoints())
+    .fill(0)
+    .map((_, i) => points.getPoint(i).slice());
+
+  const linesArray = new Array(lines.getNumberOfCells()).fill(0).map((_, i) => {
+    const cell = lines.getCell(i * 3).slice();
+    return { a: cell[0], b: cell[1] };
+  });
+
+  if (bypass) {
+    return { points: pointsArray, lines: linesArray };
+  }
+
+  const newPoints = [];
+  for (const [i, pt] of pointsArray.entries()) {
+    const index = newPoints.findIndex(
```

Comment:
This is an n^2 algorithm - consider using a Map<string,Point3>.  Generates a reasonable amount of garbage, but I think the performance of that is better than doing repeated compares across the entire array.